great_expectations/rule_based_profiler/toolkit.py

```diff
@@ -14,6 +15,65 @@
 from great_expectations.util import filter_properties_dict
 
 
+def add_profiler(
+    config: RuleBasedProfilerConfig,
+    data_context: "DataContext",  # noqa: F821
+    profiler_store: ProfilerStore,
+    ge_cloud_id: Optional[str] = None,
+) -> RuleBasedProfiler:
+    if not _check_validity_of_batch_requests_in_config(config):
+        raise ge_exceptions.InvalidConfigError(
+            f'batch_data found in batch_request cannot be saved to ProfilerStore "{profiler_store.store_name}"'
+        )
+
+    # Chetan - 20220204 - DataContext to be removed once it can be decoupled from RBP
+    # Necessary to use instantiate_class_from_config due to the possible presence of class_name/module_name in config
+    new_profiler: RuleBasedProfiler = instantiate_class_from_config(
+        config=config.to_json_dict(),
+        runtime_environment={
+            "data_context": data_context,
+        },
+        config_defaults={
+            "module_name": "great_expectations.rule_based_profiler",
+        },
+    )
```

Comment:
@donaldheppner we use `instantiate_class_from_config` here because when reading from a YAML configuration, we have `class_name` and `module_name` in the config.

We could manually pop those values from the resulting dict if we'd like (both ways don't feel too great).

Thoughts?

Reply:
I think that unless we explicitly support indirection/substitution of the `RuleBasedProfiler` type (and I don't think we do), I think we can both remove `class_name` and `module_name` from the YAML and not use `instantiate_class_from_config`.